File: experiment_jp_test/j2pyrawpy/if__1__COUNT_PALINDROMIC_SUBSEQUENCE_GIVEN_STRING_1.py
#!/usr/bin/env python
""" generated source for module if__1__COUNT_PALINDROMIC_SUBSEQUENCE_GIVEN_STRING_1 """
class X(object):
    """ generated source for class X """
    @classmethod
    def countPS(cls, i, j):
        """ generated source for method countPS """
        if (i - j == 1) or (i - j == -1):
            if str_.charAt(i) == str_.charAt(j):
                return dp[i][j] = 3
            else:
                return dp[i][j] = 2
        if i == j:
            return dp[1][j] = 1
        elif str_.charAt(i) == str_.charAt(j):
            return dp[i][j] = countPS(i + 1, j) + countPS(i, j - 1) + 1
        else:
            return dp[i][j] = countPS(i + 1, j) + countPS(i, j - 1) - countPS(i + 1, j - 1)

File: experiment_jp_test/mutants/if__1__COUNT_PALINDROMIC_SUBSEQUENCE_GIVEN_STRING_1.java
static int countPS ( int i , int j ) { if ( ( i - j == 1 ) || ( i - j == - 1 ) ) { if ( str . charAt ( i ) == str . charAt ( j ) ) return dp [ i ] [ j ] = 3 ; else return dp [ i ] [ j ] = 2 ; } if ( i == j ) return dp [ 1 ] [ j ] = 1 ; else if ( str . charAt ( i ) == str . charAt ( j ) ) return dp [ i ] [ j ] = countPS ( i + 1 , j ) + countPS ( i , j - 1 ) + 1 ; else return dp [ i ] [ j ] = countPS ( i + 1 , j ) + countPS ( i , j - 1 ) - countPS ( i + 1 , j - 1 ) ; }
